node/src/node/isometrics.cpp

```diff
@@ -0,0 +1,528 @@
+            const auto& name = init_metrics.first;
+            const auto& must_be_preserved = init_metrics.second;
+
+            dbg("preserved " << name << " : " << must_be_preserved);
+
+            auto self_it = this->common_counters.find(name);
+            if (self_it != std::end(this->common_counters) && must_be_preserved == false) {
+                    self_it->second.value->store(0);
+            }
+        }
+
+    } else {
+
+        for(const auto& proxy_metrics : proxy.common_counters) {
+            const auto& name = proxy_metrics.first;
+            const auto& proxy_record = proxy_metrics.second;
+
+            auto self_it = this->common_counters.find(name);
+            if (self_it != std::end(this->common_counters)) {
+                auto& self_record = self_it->second;
+
+                if (self_record.is_accumulated) { // monotonically increasing
+                    self_record.value->fetch_add(proxy_record.deltas);
+                } else {
+                    self_record.value->store(proxy_record.values);
+                }
+            }
+
+        } // for metrics in proxy
+    }
+
+    return *this;
+}
+
+// TODO: error messages processing
+struct response_processor_t {
+    using stats_table_type = metrics_retriever_t::stats_table_type;
+
+    struct error_t {
+        long code;
+        std::string message;
+    };
+
+    response_processor_t(const std::string app_name) :
+        app_name(app_name),
+        processed{false}
+    {}
+
+    auto
+    operator()(context_t& ctx, const dynamic_t& response, stats_table_type& stats_table) -> size_t {
+        return process(ctx, response, stats_table);
+    }
+
+    auto
+    process(context_t& ctx, const dynamic_t& response, stats_table_type& stats_table) -> size_t {
+        const auto apps = response.as_object();
+
+        auto uuids_processed = size_t{};
+
+        for(const auto& app : apps) {
+
+            // TODO: restore app_name assertion check
+            //
+            // strictly, we should have one app at all within current protocol
+            // if (app_name != app.first) {
+            //     dbg("JSON app is " << app.first);
+            //     continue;
+            // }
+
+            const auto uuids = app.second.as_object();
+
+            for(const auto& item : uuids) {
+                const auto& uuid = item.first;
+                const auto& metrics = item.second.as_object();
+
+                if (uuid.empty()) {
+                    dbg("[response] 'uuid' value is empty, ignoring");
+                    continue;
+                }
+
+                if (uuid.compare("error") == 0) {
+                    parse_error_record(metrics);
+                    continue;
+                }
+
+                auto stat_it = stats_table.find(uuid);
+                if (stat_it == std::end(stats_table)) {
+                    // If isolate daemon sends us uuid we don't know about,
```

Comment:
Well `update stamp` checking will be implemented either on this or on next iteration, stay tuned.